src/main/kotlin/com/criptext/mail/scenes/emaildetail/workers/ReadEmailsWorker.kt

```diff
@@ -36,7 +36,7 @@ class ReadEmailsWorker(private val dao: EmailDao,
         val threadIds = dao.getThreadIdsFromEmailIds(emailIds)
         val result = Result.of { apiClient.postOpenEvent(metadataKeys) }
                 .map { apiClient.postThreadReadChangedEvent(threadIds,false) }
-                .flatMap { Result.of { dao.toggleRead(ids = emailIds, unread = false) } }
+                .flatMap { Result.of { dao.toggleCheckingRead(ids = emailIds, unread = false) } }
 
```

Comment:
if you are fixing bugs in workers, you should write tests that prove that it is fixed and ensures that it remains fixed

Reply:
Now the controller don-t filter the unread emails to send to the worker, so now the Unit test for this is updated to just know if the read email worker is effectively run after LoadEmailsWorker. For knowing  if it changes specifically those emails, I will upload later today a second PR with an android test for it. As well as all android tests fixed since with the convention of always checking the http api return first, the Android tests are not working properly.